app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+            await withController(
+              {
+                state: {
+                  provider: {
+                    type: networkType,
+                  },
+                },
+              },
+              async ({ controller, network }) => {
+                network.mockEssentialRpcCalls({
+                  eth_blockNumber: {
+                    response: {
+                      result: '0x42',
+                    },
+                  },
+                });
+                await withoutCallingLookupNetwork({
+                  controller,
+                  operation: async () => {
+                    await controller.initializeProvider();
+                  },
+                });
+
+                const infuraIsUnblocked = await waitForEvent({
+                  controller,
+                  eventName: 'infuraIsUnblocked',
+                  operation: async () => {
+                    await controller.lookupNetwork();
+                  },
+                });
+
+                expect(infuraIsUnblocked).toBe(true);
+              },
+            );
+          });
+
+          it('does not emit infuraIsUnblocked if the network has changed by the time the request ends', async () => {
+            await withController(
+              {
+                state: {
+                  provider: {
+                    type: networkType,
+                  },
+                },
+              },
+              async ({ controller }) => {
+                const network1 = new NetworkCommunications({
+                  networkClientType: 'infura',
+                  infuraNetwork: networkType,
+                });
+                network1.mockEssentialRpcCalls({
+                  eth_blockNumber: {
+                    response: {
+                      body: async () => {
+                        await waitForEvent({
+                          controller,
+                          eventName: 'networkDidChange',
+                          operation: async () => {
+                            await withoutCallingLookupNetwork({
+                              controller,
+                              operation: () => {
+                                controller.setProviderType(
+                                  INFURA_NETWORKS.filter(
+                                    (network) =>
+                                      network.networkType !== networkType,
+                                  )[0],
+                                );
+                              },
+                            });
+                          },
+                        });
+                        return {
+                          result: '0x42',
+                        };
+                      },
+                    },
+                  },
+                });
+                const network2 = new NetworkCommunications({
```

Comment:
It's unclear what this was intended to do. `network2` isn't referenced anywhere, and the test passes with this commented out

Reply:
Oh I see, this was setup with the expectation that we'd be switching to a custom RPC endpoint. But the code above switches to the next infura network. Maybe a copy+paste mistake

Reply:
So this test is a little complicated. In the implementation, in `_checkInfuraAvailability`, we make a request to `eth_blockNumber`. If the network has switched, however, after the request finishes, then we don't emit any events. So `network2` is the second network that we switch to.

Reply:
> So network2 is the second network that we switch to.

That's what I mean - it's _not_ the network being switched to here. It seems like you meant it to be though.

Reply:
Derp, you're right. I don't know what I was seeing 😅 Fixed in [884e702](https://github.com/MetaMask/metamask-extension/pull/17498/commits/884e7022a6e50030fefd1df52db7b99991b15952).